src/prefect/engine/cloud/flow_runner.py

```diff
@@ -130,7 +145,8 @@ def call_runner_target_handlers(self, old_state: State, new_state: State) -> Sta
             state.state = old_state  # type: ignore
             raise ENDRUN(state=state)
 
-        prefect.context.update(flow_run_version=version + 1)
+        if version is not None:
```

Comment:
This is actually not quite right - the version will still increment in the server, so Core should continue incrementing the version even if it's not providing it. The reason for this is that if the version isn't always incremented, then clients that DO want to enforce locking can't actually tell if they have the latest version or not.